stdlib/3/optparse.pyi

```diff
@@ -0,0 +1,248 @@
+# Generated by pytype, with only minor tweaks. Might be incomplete.
+# Updated for Python3.
+
+from typing import Any, Optional, List, Callable, Tuple, Dict, Iterable, Union
+
+
+NO_DEFAULT = ...  # type: Tuple[str, ...]
+SUPPRESS_HELP = ...  # type: str
+SUPPRESS_USAGE = ...  # type: str
+
+def check_builtin(option: Option, opt, value: str) -> Any: ...
+def check_choice(option: Option, opt, value) -> Any: ...
+def isbasestring(x) -> bool: ...
+
+class OptParseError(Exception):
+    msg = ...  # type: Any
+    def __init__(self, msg) -> None: ...
+
+class BadOptionError(OptParseError):
+    opt_str = ...  # type: Any
+    def __init__(self, opt_str) -> None: ...
+
+class AmbiguousOptionError(BadOptionError):
+    possibilities = ...  # type: Any
+    def __init__(self, opt_str, possibilities) -> None: ...
+
+class OptionError(OptParseError):
+    msg = ...  # type: Any
+    option_id = ...  # type: str
+    def __init__(self, msg, option: Option) -> None: ...
+
+class OptionConflictError(OptionError): ...
+
+class HelpFormatter:
+    NO_DEFAULT_VALUE = ...  # type: str
+    _long_opt_fmt = ...  # type: str
+    _short_opt_fmt = ...  # type: str
+    current_indent = ...  # type: int
+    default_tag = ...  # type: str
+    help_position = ...  # type: Any
+    help_width = ...  # type: Any
+    indent_increment = ...  # type: Any
+    level = ...  # type: int
+    max_help_position = ...  # type: int
+    option_strings = ...  # type: Dict[Option, str]
+    parser = ...  # type: Any
```

Comment:
OptionParser according to the docstring. A bunch of other attributes also have more concrete types according to the docstring.